### PR TITLE
WIP: Feature/stats server latency

### PR DESCRIPTION
**Pull request by git-hulk**
- Add server latency metrics
- Add request latency metrics

- [x] TODO: Benchmark and check latency

### CODE REVIEW

File: src/nc_request.c [hunk 1]
```diff
@@ -112,6 +114,11 @@ req_put(struct msg *msg)
         msg->peer = NULL;
         pmsg->peer = NULL;
         rsp_put(pmsg);
+        /* only record the server pool message */
+        if (conn->client && !conn->proxy) {
+            req_time = nc_msec_now()-msg->start_ts/1000;
```
Comment (ruoshan) [inline src/nc_request.c]: 这边计时的话，很多错误的返回也会被计算到吧

File: src/nc_request.c [hunk 2]
```diff
@@ -112,6 +114,11 @@ req_put(struct msg *msg)
...
+            stats_pool_record_latency(conn_to_ctx(conn), conn->owner, req_time);
```
Comment (karelrooted) [inline src/nc_request.c]: memcache get command will be recorded twice in pool latency , please fix it before merging

**Review by karelrooted — DISMISSED** [review]

File: src/nc_request.c (hunk 2 above)
Comment (karelrooted) [reply src/nc_request.c]: should not record fragment msg as pool latency，add check for fragment ?
            if (msg->frag_id == 0 || msg->frag_owner == msg) {

File: src/nc_request.c (hunk 1 above)
Comment (git-hulk) [reply src/nc_request.c]: Fixed

File: src/nc_request.c (hunk 2 above)
Comment (git-hulk) [reply src/nc_request.c]: Fixed

**Review by ruoshan — COMMENTED** [review]
LGTM

File: src/nc_request.c [hunk 3]
```diff
@@ -101,6 +101,7 @@ void
 req_put(struct msg *msg)
 {
     struct msg *pmsg; /* peer message (response) */
+    struct conn *conn = msg->owner;
```
Comment (karelrooted) [inline src/nc_request.c]: warning: introduce a unused variable here

Comment (git-hulk) [reply src/nc_request.c]: Fixed

**Review by ruoshan — APPROVED** [review]

**Review by karelrooted — APPROVED** [review]

**Pull request merged**